pilot/pkg/bootstrap/istio_ca.go

```
@@ -263,13 +263,23 @@ func (s *Server) loadCACerts(caOpts *caOptions, dir string) error {
 		return nil
 	}
 
-	signingKeyFile := path.Join(dir, ca.CAPrivateKeyFile)
-	if _, err := os.Stat(signingKeyFile); err == nil {
+	signingCABundleComplete, bundleExists, err := checkCABundleCompleteness(
+		path.Join(dir, ca.CAPrivateKeyFile),
+		path.Join(dir, ca.CACertFile),
+		path.Join(dir, ca.RootCertFile),
+		[]string{path.Join(dir, ca.CertChainFile)},
+	)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("error loading remote CA certs: %w", err)
+	}
+	if signingCABundleComplete {
 		return nil
-	} else if !os.IsNotExist(err) {
-		return fmt.Errorf("signing key file %s already exists", signingKeyFile)
+	}
+	if bundleExists {
+		log.Warnf("incomplete signing CA bundle detected at %s", dir)
```

Do we want to allow a silent fallback to using an external bundle?

---

I only added a warning here since the next block tries to load it up from the remote cluster, which is where we fail with an error if need be. fallback seems intentional here?

---

This PR is addressing not silently falling back to using a self-signed CA when an incomplete ca bundle exists. Do you think there is a scenario where is a user is expecting to use a local ca bundle if it exists, but ends up using a remote ca bundle since the local one is incomplete? Not sure what the right answer is here

---

function description comments does say that we want to fetch the ca bundle from remote cluster instead. Hence I kept the existing behavior intact with a warning message.

`loadCACerts loads an existing cacerts Secret if the files aren't mounted locally.
 By default, a cacerts Secret would be mounted during pod startup due to the
 Istiod Deployment configuration. But with external Istiod, we want to be
 able to load cacerts from a remote cluster instead.`